### PR TITLE
refactor: update to use pydantic refactor core ape

### PR DESCRIPTION
**Pull request by NotPeopling2day**
### What I did

fixes: #

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_trezor/accounts.py [hunk 1]
```diff
@@ -25,9 +25,10 @@ def aliases(self) -> Iterator[str]:
     def __len__(self) -> int:
         return len([*self._account_files])
 
-    def __iter__(self) -> Iterator[AccountAPI]:
+    @property
+    def accounts(self) -> Iterator[AccountAPI]:
         for account_file in self._account_files:
-            yield TrezorAccount(self, account_file)  # type: ignore
+            yield TrezorAccount(container=self, account_file_path=account_file)  # type: ignore
```
Comment (fubuloubu) [inline ape_trezor/accounts.py:31]: Is container injected?

**Review by fubuloubu — APPROVED** [review]

File: ape_trezor/accounts.py (hunk 1 above)
Comment (NotPeopling2day) [reply ape_trezor/accounts.py]: You are correct!

**Comment by NotPeopling2day** [comment]
This PR can be merged after https://github.com/ApeWorX/ape/pull/509 is merged into core.

File: ape_trezor/client.py [hunk 2]
```diff
@@ -85,31 +85,27 @@ def sign_personal_message(self, message: bytes) -> Tuple[int, bytes, bytes]:
         to validate the message data.
         """
         ethereum_message_signature = ethereum.sign_message(
-            self._client, parse_hdpath(self._account_hd_path.path), message
+            self.client, parse_hdpath(self._account_hd_path.path), message
         )
 
         return extract_signature_vrs_bytes(signature_bytes=ethereum_message_signature.signature)
 
-    # TODO: Uncomment when Trezor has released the EIP 712 update
```
Comment (unparalleled-js) [inline ape_trezor/client.py:93]: if this is released now, I would expect us to have to bump the version number of `trezor` in the `setup.py` to be able to use it?

**Review by fubuloubu — APPROVED** [review]

**Review by unparalleled-js — APPROVED** [review]
LGTM

**Pull request merged**